File: utilities/numbers/python/basic_utils.py
def clamp(value, minimum, maximum):
    return min(max(value, minimum), maximum)


def is_numeric(value):
    if value == "":
        return False
    try:
        float(value)
        return True
    except (TypeError, ValueError):
        return False


def min_value(values):
    return min(values) if values else None


def max_value(values):
    return max(values) if values else None

File: utilities/numbers/typescript/basicUtils.ts
export const clamp = (value: number, min: number, max: number): number => Math.min(Math.max(value, min), max);

export const isNumeric = (value: unknown): boolean => value !== "" && Number.isFinite(Number(value));

export const minValue = (values: number[]): number | undefined => values.length === 0 ? undefined : Math.min(...values);

export const maxValue = (values: number[]): number | undefined => values.length === 0 ? undefined : Math.max(...values);
